gui/ThemeParser.cpp

```diff
@@ -740,7 +740,18 @@ bool ThemeParser::parserCallback_dialog(ParserNode *node) {
 			return false;
 	}
 
-	_theme->getEvaluator()->addDialog(name, node->values["overlays"], enabled, inset);
+	Common::String overlays = node->values["overlays"];
+
+	int width = -1, height = -1;
+	if (node->values.contains("size")) {
+		if (overlays != "screen_center")
+			return parserError("Dialogs can only have an explicit size if they overlay 'screen_center'.");
```

Comment:
Could you add a comment here, explaining why?

Reply:
Okay.